src/main/java/commission/service/CommissionServiceImpl.java

```diff
@@ -0,0 +1,63 @@
+package commission.service;
+
+import commission.dao.CrudDao;
+import commission.entity.Commission;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class CommissionServiceImpl implements CrudService<Commission> {
+    public final CrudDao<Commission> commissionDao;
+    private final Log logger = LogFactory.getLog(this.getClass());
+
+
+    @Autowired
+    public CommissionServiceImpl(CrudDao<Commission> commissionDao) {
+        this.commissionDao = commissionDao;
+    }
+
+    @Override
+    public List<Commission> findAll() {
+        return commissionDao.findAll();
+    }
+
+    @Override
+    public void insert(Commission commission) {
+        commissionDao.insert(commission);
+        logger.info("Commission created");
+    }
+
+    @Override
+    public Optional<Commission> findById(long id) {
+        var temp = commissionDao.findById(id);
+        if (temp.isPresent()){
+            return  temp;
+        } else {
+            logger.error("ID " + id + " not found.");
+        }
+        return Optional.empty();
```

Comment:
You are returning in the `if` part of the condition, in my mind it reads easier if you also then return in the `else` part.

You could also restructure the code to read easier like below. Then we don't have to return `Optional.empty()` explicitly as we just return whatever the value of `temp` is, since the only reason for that `if` statement is to log when it's empty
```suggestion
        var temp = commissionDao.findById(id);
        if (temp.isEmpty()){
            logger.error("ID " + id + " not found.");
        }
        return temp;
```

Reply:
Yes it is much more clear.  Applied this change across the code.